qemu/tests/in_place_upgrade_legacy.py

```diff
@@ -0,0 +1,126 @@
+    4.pre_upgrade test in the vm
+    5.upgrade test in the vm
+    6.check if it's target system
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    upgrade_test = IpuLegacyTest(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    upgrade_test.session = vm.wait_for_login(timeout=login_timeout)
+    try:
+        # set post_release
+        pre_release = params.get("pre_release")
+        release_chk = params.get("release_check")
+        if pre_release not in upgrade_test.run_guest_cmd(release_chk):
+            test.cancel("your image is not for rhel 7.9 product, please check")
+        post_release = params.get("post_release")
+        if params.get("rhsm_type") == "no_rhsm":
+            # internal repo for doing yum update
+            old_custom_repo = params.get("old_custom_internal_repo_7")
+            upgrade_test.yum_update_no_rhsm(test, old_custom_repo)
+        elif params.get("rhsm_type") == "rhsm":
+            # doing rhsm and update the old system
+            upgrade_test.run_guest_cmd(params.get("prepare_env"))
+            upgrade_test.run_guest_cmd(params.get("get_answer_files_source"))
+            upgrade_test.run_guest_cmd(params.get("get_answer_files"))
+            upgrade_test.rhsm(test)
+        upgrade_test.session = vm.reboot(upgrade_test.session)
+        upgrade_test.run_guest_cmd(params.get("repo_leapp_7"))
+        upgrade_test.run_guest_cmd(params.get("repo_leppp_7_seed"))
+        upgrade_test.run_guest_cmd(params.get("ins_leapp_cmd_7"))
+        if params.get("rhsm_type") == "no_rhsm":
+            upgrade_test.run_guest_cmd(params.get("prepare_env"))
+            upgrade_test.run_guest_cmd(params.get("get_answer_files_source"))
+            upgrade_test.run_guest_cmd(params.get("get_answer_files"))
+            upgrade_test.run_guest_cmd(params.get("get_custom_7"))
+            upgrade_test.run_guest_cmd(params.get("export_type_7"))
+            upgrade_test.run_guest_cmd(params.get("new_rhel_content_7"))
+        upgrade_test.pre_upgrade_whitelist(test)
+        if params.get("rhsm_type") == "no_rhsm":
+            upgrade_test.run_guest_cmd(params.get("pre_upgrade_no_rhsm"))
+            upgrade_test.upgrade_process(params.get("process_upgrade_no_rhsm"))
+        elif params.get("rhsm_type") == "rhsm":
+            upgrade_test.run_guest_cmd(params.get("leapp_proxy_host"))
+            upgrade_test.run_guest_cmd(params.get("leapp_service_host"))
+            upgrade_test.run_guest_cmd(params.get("pre_upgrade_rhsm"))
+            upgrade_test.upgrade_process(params.get("process_upgrade_rhsm"))
+        try:
+            upgrade_test.session.cmd("reboot")
+        except aexpect.ShellProcessTerminatedError:
+            pass
+        # post checking
+        upgrade_test.session = vm.wait_for_login(timeout=6000)
```

Comment:
Why not `vm.reboot(timeout=6000)`

Reply:
The test steps are like "leapp upgrade ..." and "reboot" so I followed them strictly but if it's fine for me to replace it with vm.reboot(). Thanks.

Reply:
Updated it to session.sendline().